homeassistant/components/guardian/__init__.py

```diff
@@ -142,50 +255,131 @@ def icon(self) -> str:
         """Return the icon."""
         return self._icon
 
+    @callback
+    def _async_update_from_latest_data(self):
+        """Update the entity.
+
+        This should be extended by Guardian platforms.
+        """
+        raise NotImplementedError
+
+    @callback
+    def _async_update_state_callback(self):
+        """Update the entity's state."""
+        self._async_update_from_latest_data()
+        self.async_write_ha_state()
+
+
+class PairedSensorEntity(GuardianEntity):
+    """Define a Guardian paired sensor entity."""
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        coordinator: DataUpdateCoordinator,
+        kind: str,
+        name: str,
+        device_class: str,
+        icon: str,
+    ) -> None:
+        """Initialize."""
+        super().__init__(entry, kind, name, device_class, icon)
+
+        self.coordinator = coordinator
+
+        self._paired_sensor_uid = coordinator.data["uid"]
+
+        self._device_info["identifiers"] = {(DOMAIN, self._paired_sensor_uid)}
+        self._device_info["name"] = f"Guardian Paired Sensor {self._paired_sensor_uid}"
+        self._device_info["via_device"] = (DOMAIN, self._entry.data[CONF_UID])
+
     @property
     def name(self) -> str:
         """Return the name of the entity."""
-        return f"Guardian {self._valve_controller_uid}: {self._name}"
+        return f"Guardian Paired Sensor {self._paired_sensor_uid}: {self._name}"
 
     @property
-    def should_poll(self) -> bool:
-        """Return True if entity has to be polled for state."""
-        return False
+    def unique_id(self):
+        """Return the unique ID of the entity."""
+        return f"{self._paired_sensor_uid}_{self._kind}"
+
+    async def async_added_to_hass(self) -> None:
+        """Perform tasks when the entity is added."""
+        await super().async_added_to_hass()
```

Comment:
The CoordinatorEntity will register a default update callback here. If we're not going to use that we should not register it.